bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/DefaultSystemChannelTypeProvider.java

```diff
@@ -89,34 +100,107 @@
                     new EventOption(CommonTriggerEvents.LONG_PRESSED, null))),
             null);
 
+    private class LocalizedChannelTypeKey {
```

Comment:
Could you make this a static class? Then I think you also don't need to check for the outer type in hashcode and equals.